code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -217,3 +218,39 @@
 	name = "resting contraption"
 	desc = "This looks similar to contraptions from Earth. Could aliens be stealing our technology?"
 	icon_state = "abed"
+
+//unfortunateley no sickness mechanics on them... yet
+/obj/structure/bed/maint
+	name = "dirty mattress"
+	desc = "An old grubby mattress. You try to not think about what could be the cause of those stains."
+	icon_state = "dirty_mattress"
+
+//Double Beds, for luxurious sleeping, i.e. the captain and maybe heads - if people use this for ERP, send them to skyrat, or worse, acacia
+/obj/structure/bed/double
+	name = "double bed"
+	desc = "A luxurious double bed, for those too important for small dreams."
+	icon_state = "bed_double"
+	buildstackamount = 4
+	max_buckled_mobs = 2
+	///The mob who buckled to this bed second, to avoid other mobs getting pixel-shifted before they unbuckles.
+	var/mob/living/goldilocks
```

Comment:
possible hard-del.

Register a QDELETING signal - on qdel set to null and reset the pixel height

Reply:
could also use a weakref

https://hackmd.io/@PowerfulBacon/guide_to_harddels

Reply:
leaving this open because not sure if i fixed it?